tests/common/include/tests/common/tools_line_intersection.inl

```diff
@@ -0,0 +1,157 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Project include(s)
+#include "detray/intersection/intersection.hpp"
+#include "detray/intersection/ray_line_intersector.hpp"
+#include "detray/masks/line.hpp"
+#include "detray/propagator/track.hpp"
+
+// GTest include(s)
+#include <gtest/gtest.h>
+
+// System include(s)
+#include <climits>
+#include <cmath>
+
+/// @note __plugin has to be defined with a preprocessor command
+using namespace detray;
+
+// Three-dimensional definitions
+using cartesian = __plugin::cartesian2<detray::scalar>;
+using transform3 = __plugin::transform3<detray::scalar>;
+using vector3 = __plugin::vector3<scalar>;
+using point3 = __plugin::point3<scalar>;
+using point2 = __plugin::point2<scalar>;
+constexpr scalar tolerance = 1e-5;
+
+// This defines the local frame test suite
+TEST(tools, line_intersector) {
+
+    // tf3 with Identity rotation and no translation
+    vector3 x{1, 0, 0};
+    vector3 y{0, 1, 0};
+    vector3 z{0, 0, 1};
+    vector3 t{0, 0, 0};
+
+    const transform3 tf{t, x, y, z};
+
+    /// Create a track
+    std::vector<free_track_parameters> trks(2);
+    trks[0] = free_track_parameters({1, -1, 0}, 0, {0, 1, 0}, -1);
+    trks[1] = free_track_parameters({-1, -1, 0}, 0, {0, 1, 0}, -1);
+
+    // Infinite wire with 10 mm radial cell size
+    line<> ln{10., std::numeric_limits<scalar>::infinity(), 0u};
+
+    // Test intersect
+    std::vector<ray_line_intersector::intersection_type> is(2);
```

Comment:
Btw, is there a particular reason to use vectors here?

Reply:
I thought it's easier to add cases